src/Raven.Client/Util/StreamWithTimeout.cs

```diff
@@ -26,24 +32,93 @@ internal sealed class StreamWithTimeout : Stream
         private CancellationToken _requestWriteCts;
 #endif
 
-        private long _totalRead;
-        private long _totalWritten;
+        private long _totalRead = 0;
+        private long _totalWritten = 0;
 
         public StreamWithTimeout(Stream stream)
         {
             _stream = stream;
+            SetWriteTimeoutIfNeeded(DefaultWriteTimeout);
+            SetReadTimeoutIfNeeded(DefaultReadTimeout);
```

Comment:
The issue here is that this forces us to setup a timer , which is what we wanted to avoid, no?